### PR TITLE
sysz: init at 1.2.2

### LINKED ISSUE #138472: Package Request:  sysz - an fzf terminal UI for systemctl
**Project description**

A terminal UI utility for using systemctl interactively via fzf.

**Metadata**

* homepage URL: https://github.com/joehillen/sysz
* source URL: https://github.com/joehillen/sysz
* license: Unlicense
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by thbkrshw**
###### Motivation for this change

Add a new package

Closes #138472

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thbkrshw** [comment]
Fix #138472

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4431,6 +4431,12 @@
     githubId = 5525217;
     name = "Hunter Jones";
   };
+  hleboulanger = {
+    email = "hleboulanger@protonmail.com";
+    name = "Harold Leboulanger";
+    github = "thbkrhsw";
+    githubId = 33122;
+  };
```
Comment (NerdyPepper) [inline maintainers/maintainer-list.nix:4439]: Please put this into a separate commit with the message `maintainers: add NAME` and sort it alphabetically

File: pkgs/tools/misc/sysz/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ stdenv,  fetchFromGitHub, lib, bash, gawk, fzf }:
```
Comment (NerdyPepper) [inline pkgs/tools/misc/sysz/default.nix:1]:
```suggestion
{ stdenvNoCC, fetchFromGitHub, lib, bash, gawk, fzf }:
```

File: pkgs/tools/misc/sysz/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+stdenv.mkDerivation rec {
```
Comment (NerdyPepper) [inline pkgs/tools/misc/sysz/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

File: pkgs/tools/misc/sysz/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  preBuild = "patchShebangs README.sh";
+
+  build = "
+    make all
+  ";
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp sysz $out/bin
```
Comment (NerdyPepper) [inline pkgs/tools/misc/sysz/default.nix:28]:
```suggestion
    install -Dm755 -t $out/bin sysz
```

**Review by oppiliappan — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/sysz/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    gawk
+    fzf
+  ];
+
+  preBuild = "patchShebangs README.sh";
+
+  build = "
+    make all
+  ";
+
```
Comment (figsoda) [inline pkgs/tools/misc/sysz/default.nix:25]: the makefile doesn't do anything to the binary
```suggestion
```

File: pkgs/tools/misc/sysz/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    rev = version;
+    sha256 = "sha256:0idi94jb075jki1jkm7shmicv77ylm2xjpjq6c868j9w6pvnash0";
+  };
+
+  buildInputs = [
+    bash
+    gawk
+    fzf
+  ];
+
```
Comment (figsoda) [inline pkgs/tools/misc/sysz/default.nix:19]: fzf and gawk should be added with `makeWrapper` or `wrapProgram`, adding them to buildInputs doesn't do anything
```suggestion
```

File: pkgs/tools/misc/sysz/default.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  preBuild = "patchShebangs README.sh";
+
+  build = "
+    make all
+  ";
+
+  installPhase = ''
+    install -Dm755 -t $out/bin sysz
+  '';
```
Comment (figsoda) [inline pkgs/tools/misc/sysz/default.nix:28]: use `makeWrapper` or `wrapProgram` so `fzf` and `awk` are in `PATH` at runtime

**Review by figsoda — CHANGES_REQUESTED** [review]
`maintainers: add hleboulanger` commit should be before `sysz:init at 1.2.2` commit, and they should be the only two commits

**Comment by thbkrshw** [comment]
I kept the Makefile because `SYSZ_VERSION` is replaced in the `all` target.

File: pkgs/tools/misc/sysz/default.nix [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "joehillen";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:0idi94jb075jki1jkm7shmicv77ylm2xjpjq6c868j9w6pvnash0";
+  };
+
+  buildInputs = [ makeWrapper ];
+  preBuild = "patchShebangs README.sh";
+  build = "make all";
```
Comment (figsoda) [inline pkgs/tools/misc/sysz/default.nix:16]: SYSZ_VERSION is up to date in the `sysz` file, i think the makefile is only meant for the developers

```suggestion

dontBuild = true;
```

File: pkgs/tools/misc/sysz/default.nix [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    makeWrapper sysz $out/bin/sysz --prefix PATH : ${lib.makeBinPath [ fzf gawk ]}
+  '';
```
Comment (figsoda) [inline pkgs/tools/misc/sysz/default.nix:21]:
```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm755 sysz $out/libexec/sysz
    makeWrapper $out/libexec/sysz $out/bin/sysz \
      --prefix PATH : ${lib.makeBinPath [ fzf gawk ]}
    runHook postInstall
  '';
```

File: pkgs/tools/misc/sysz/default.nix [hunk 10]
```diff
@@ -0,0 +1,31 @@
+{ stdenvNoCC,  fetchFromGitHub, lib, makeWrapper, bash, gawk, fzf }:
```
Comment (figsoda) [inline pkgs/tools/misc/sysz/default.nix:1]:
```suggestion
{ lib, stdenvNoCC, fetchFromGitHub, makeWrapper, fzf, gawk }:
```

**Review by figsoda — CHANGES_REQUESTED** [review]

**Comment by figsoda** [comment]
@ofborg build sysz

File: pkgs/tools/misc/sysz/default.nix [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+stdenvNoCC.mkDerivation rec {
+  pname = "sysz";
+  version = "1.2.2";
+
+  src = fetchFromGitHub {
+    owner = "joehillen";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256:0idi94jb075jki1jkm7shmicv77ylm2xjpjq6c868j9w6pvnash0";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sysz/default.nix:11]:
```suggestion
    sha256 = "0idi94jb075jki1jkm7shmicv77ylm2xjpjq6c868j9w6pvnash0";
```

File: pkgs/top-level/all-packages.nix [hunk 12]
```diff
@@ -32764,4 +32764,6 @@ with pkgs;
   zthrottle = callPackage ../tools/misc/zthrottle { };
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  sysz = callPackage  ../tools/misc/sysz { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:32768]: Please sort this into a fitting category and sort there alphabetically.

**Comment by FlorianFranzen** [comment]
@thbkrshw Thank you for getting this going. Looks like [v1.3.0](https://github.com/joehillen/sysz/releases/tag/1.3.0) was release in the meantime.

**Comment by thbkrshw** [comment]
@FlorianFranzen should be nearly good now ;)

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 138507` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sysz</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]
